src/meta/src/hummock/manager/compaction.rs

```diff
@@ -265,6 +266,67 @@ impl HummockManager {
             } => {}
         }
     }
+
+    pub(crate) async fn calculate_vnode_partition(
+        &self,
+        compact_task: &mut CompactTask,
+        compaction_config: &CompactionConfig,
+    ) {
+        if compaction_config.split_weight_by_vnode > 0 {
+            for table_id in &compact_task.existing_table_ids {
+                compact_task
+                    .table_vnode_partition
+                    .insert(*table_id, compact_task.split_weight_by_vnode);
+            }
+        } else {
+            let mut table_size_info: HashMap<u32, u64> = HashMap::default();
+            let mut existing_table_ids: HashSet<u32> = HashSet::default();
+            for input_ssts in &compact_task.input_ssts {
+                for sst in &input_ssts.table_infos {
+                    existing_table_ids.extend(sst.table_ids.iter());
+                    if !sst.table_ids.is_empty() {
+                        *table_size_info.entry(sst.table_ids[0]).or_default() +=
```

Comment:
Why only table_ids[0] is counted, is this an estimation algorithm?

Reply:
Emmm, No. I will fix it. In fact, I split all sst by table just to make here calculation to be accurately